tests/publiccloud/run_ltp.pm

```diff
@@ -152,42 +195,50 @@ sub run {
     } else {
         record_info("Exclude", "None");
     }
+    return $skip_tests;
+}
 
+sub prepare_kirk {
+    my ($self, $instance) = @_;
     my $kirk_repo = get_var("LTP_RUN_NG_REPO", "https://github.com/linux-test-project/kirk.git");
     my $kirk_branch = get_var("LTP_RUN_NG_BRANCH", "master");
     record_info('LTP RUNNER REPO', "Repo: " . $kirk_repo . "\nBranch: " . $kirk_branch);
     script_retry("git clone -q --single-branch -b $kirk_branch --depth 1 $kirk_repo", retry => 5, delay => 60, timeout => 300);
     $instance->run_ssh_command(cmd => 'sudo CREATE_ENTRIES=1 ' . get_ltproot() . '/IDcheck.sh', timeout => 300);
     record_info('Kernel info', $instance->run_ssh_command(cmd => q(rpm -qa 'kernel*' --qf '%{NAME}\n' | sort | uniq | xargs rpm -qi)));
     if (get_var('PUBLIC_CLOUD_INSTANCE_TYPE') =~ /-metal$/) {
-        # The metal detector fails on GCP because it may return "google"
         record_info('VM type', $instance->run_ssh_command(cmd => '! systemd-detect-virt')) unless is_gce;
     } else {
         record_info('VM type', $instance->run_ssh_command(cmd => 'systemd-detect-virt'));
     }
+    assert_script_run("cd kirk");
+    my $ghash = script_output("git rev-parse HEAD", proceed_on_failure => 1);
+    set_var("LTP_RUN_NG_GIT_HASH", $ghash);
+    record_info("KIRK_GIT_HASH", "$ghash");
+    assert_script_run("python3.11 -m venv env311");
+    assert_script_run("source env311/bin/activate");
+    assert_script_run("pip3.11 install asyncssh msgpack");
+}
 
+sub upload_runtest {
+    my ($self, $instance, $provider) = @_;
     assert_script_run('curl ' . data_url('publiccloud/ltp_runtest') . ' -o publiccloud');
     $instance->scp("publiccloud", 'remote:/tmp/publiccloud', 9999);
     $instance->ssh_assert_script_run(cmd => "sudo mv /tmp/publiccloud /opt/ltp/runtest/publiccloud");
+}
 
-    # this will print /all/ kernel messages to the console. So in case kernel panic we will have some data to analyse
+sub printk_loglevel {
+    my ($self, $instance) = @_;
     $instance->ssh_assert_script_run(cmd => "echo 1 | sudo tee /sys/module/printk/parameters/ignore_loglevel");
+}
 
-    my $reset_cmd = $root_dir . '/restart_instance.sh ' . instance_log_args($provider, $instance);
-    my $log_start_cmd = $root_dir . '/log_instance.sh start ' . instance_log_args($provider, $instance);
-
-    my $env = get_var('LTP_PC_RUNLTP_ENV');
-
+sub prepare_logging {
+    my ($self, $log_start_cmd) = @_;
     assert_script_run($log_start_cmd);
+}
 
-    assert_script_run("cd kirk");
-    my $ghash = script_output("git rev-parse HEAD", proceed_on_failure => 1);
-    set_var("LTP_RUN_NG_GIT_HASH", $ghash);
-    record_info("KIRK_GIT_HASH", "$ghash");
-    assert_script_run("python3.11 -m venv env311");
-    assert_script_run("source env311/bin/activate");
-    assert_script_run("pip3.11 install asyncssh msgpack");
-
+sub run_ltp {
```

Comment:
This is no more a running a command, but the command line composition: the name should be changed accordingly, like `prepare_ltp` or similar.